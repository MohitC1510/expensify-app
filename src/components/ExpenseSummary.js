import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpenseTotal from '../selectors/Expense-Total';

export const ExpenseSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense': 'expenses';
    const formattedAmount=numeral(expensesTotal/100).format('$0,0.00');
       return (
          <div>
              <h1>Viewing {expenseCount} {expenseWord} totaling to {formattedAmount}</h1>
          </div>
       );
}

const mapStateToProps = (state) => {
   const visibleExpenses = selectExpenses(state.expenses, state.filters);

   return{
       expenseCount: visibleExpenses.length,
       expensesTotal: selectExpenseTotal(visibleExpenses)
   }
};

export default connect(mapStateToProps)(ExpenseSummary)