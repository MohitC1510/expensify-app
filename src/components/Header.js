import React from 'react';
import { Link, NavLink } from "react-router-dom";
const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName='isActive' exact={true}>Go home</NavLink>
    <NavLink to="/create" activeClassName='isActive'>ADD expense</NavLink>
  </header>
);
export default Header;