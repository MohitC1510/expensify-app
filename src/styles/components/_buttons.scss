.button{
    background-color: $primary;
    border: none;
    color: floralwhite;
    display: inline-block;
    font-size: $mx-size;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}
.button--link{
    background: none;
}
.button--secondary{
    background: grey;
}