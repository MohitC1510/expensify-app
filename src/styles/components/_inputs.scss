.text-input{
    border: 1px solid #cacccd;
    height: 47px;
    font-size: $mx-size;
    font-weight: 300;
    padding: 0.2rem;
}
.select{
    @extend .text-input;

}

.textarea{
    @extend .text-input;
    height: 10rem;
}