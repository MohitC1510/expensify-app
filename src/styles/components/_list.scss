.list-header{
    background: $grey;
    border: 1px solid darken($grey, 10%);
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-body{
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint){
        margin-bottom: $l-size;
    }
}

.list-item{
    border: 1px solid darken($grey, 10%);
    border-top: none;
    color: $dark-grey;
    display: flex;
    flex-direction: column;
    text-decoration: none;
    padding: $s-size;
    transition: background 0.5s ease;
    &:hover{
        background: rgb(248, 248, 248);
    }
    @media (min-width: $desktop-breakpoint){
        align-items: center;
        justify-content: space-between;
        flex-direction: row;
        padding: $m-size;
    }
}

.list-item--message{
   align-items: center;
   justify-content: center;
   color: $dark-grey;
}

.list-item__title{
    margin: 0;
    word-break: break-all;
}


.list-item__subtitle{
    color: $dark-grey;
    font-size: 1.4rem;
}

.list-item__data{
    margin: $s-size 0 0 0;
    @media (min-width: $desktop-breakpoint){
        margin: 0;
        padding-left: $s-size;
    }
}
