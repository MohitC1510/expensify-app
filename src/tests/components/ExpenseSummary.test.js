import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('Should render expense summary with single expense', ()=>{
       const wrapper= shallow(<ExpenseSummary expenseCount={1} expensesTotal={200}/>);
       expect(wrapper).toMatchSnapshot();
})


test('Should render expense summary with multiple expenses', ()=>{
    const wrapper= shallow(<ExpenseSummary expenseCount={3} expensesTotal={15000}/>);
    expect(wrapper).toMatchSnapshot();
})