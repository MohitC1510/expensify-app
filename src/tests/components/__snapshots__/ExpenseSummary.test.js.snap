// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totaling to 
    $150.00
  </h1>
</div>
`;

exports[`Should render expense summary with single expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling to 
    $2.00
  </h1>
</div>
`;
